File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        
        if(cacheSize == 0) return cities.length *5;
        
        Deque<String> cache = new ArrayDeque<>();
        int answer = 0;
        
        for(int i=0; i<cities.length; i++){
            String now = cities[i].toLowerCase();
            if(cache.contains(now)){
                answer += 1;
                cache.remove(now);
            }else{
                if(cache.size() >= cacheSize){
                    cache.pollFirst();
                }
                answer += 5;
            }
            cache.add(now);
        }

        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    
    if(cacheSize == 0) return cities.length * 5;
    
    const cache = [];
    var answer = 0;
    
    for(let i=0; i<cities.length; i++){
        var now = cities[i].toLowerCase();
        if(cache.indexOf(now) >= 0){
            answer += 1;
            cache.splice(cache.indexOf(now), 1)
        }else{
            answer += 5
            if(cache.length >= cacheSize){
                cache.shift()
            }
        }
        
        cache.push(now)

    }
    
    return answer;
}